src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier;
+    private final Random random;
+    private final double defaultRadius = 10;
+    private final String defaultColor = Color.WHITE.name();
+    private final Figure defaultFigure = new Circle(defaultColor, defaultRadius);
+
+    public FigureSupplier() {
+        // Should I pass this random object to color supplier constructor, or it is bad practice ?
+        // And is it better to create another random class in color supplier ?
+        this.random = new Random();
+        this.colorSupplier = new ColorSupplier(random);
+    }
+
+    public Figure getRandomFigure() {
+        int maxLengthInUnits = 100;
+        double firstValue = random.nextDouble() * maxLengthInUnits;
+        double secondValue = random.nextDouble() * maxLengthInUnits;
+        double thirdValue = random.nextDouble() * maxLengthInUnits;
```

Comment:
You don`t need it here, if you create Circle or Square so for what you need second and third value and your naming not ok